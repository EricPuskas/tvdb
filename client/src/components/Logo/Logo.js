import React from "react";
import { Link } from "react-router-dom";
import "./Logo.scss";

const Logo = ({ text }) => {
  return (
    <div className="Logo">
      <Link to="/">
        <img
          src="https://res.cloudinary.com/zenipsstudio/image/upload/v1572047365/logo.png"
          alt="TVDB"
        />
      </Link>
      {text ? <h2> {text} </h2> : null}
    </div>
  );
};

export default Logo;
