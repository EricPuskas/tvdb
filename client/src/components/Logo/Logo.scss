.Logo {
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
}

/* Extra small devices (phones, 600px and down) */
@media only screen and (max-width: 600px) {
  .Logo img {
    max-width: 100%;
  }
}
