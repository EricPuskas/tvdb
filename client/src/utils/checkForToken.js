import jwt_decode from "jwt-decode";

// Redux Store
import store from "../store";

// Utilities
import setToken from "./setToken";

// Actions
import { getToken } from "../actions/token";

const checkForToken = () => {
  // Check for token
  if (!localStorage.token) {
    store.dispatch(getToken());
    setToken(localStorage.token);
  } else {
    setToken(localStorage.token);
    // Decode token to be able to check if it expired
    const decoded = jwt_decode(localStorage.token);
    // Check for expired token
    const currentTime = Date.now() / 1000;
    if (decoded.exp < currentTime) {
      store.dispatch(getToken());
    }
  }
};

export default checkForToken;
