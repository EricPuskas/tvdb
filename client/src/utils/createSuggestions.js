const filterInput = input => {
  return input.replace(/[^\w\s]/gi, "");
};

const createSuggestions = (word, data) => {
  let filter = filterInput(word);
  const re = new RegExp(`${filter.toLowerCase()}.*\\B`, "g");
  return data.filter(item => re.test(item.toLowerCase()));
};

export default createSuggestions;
