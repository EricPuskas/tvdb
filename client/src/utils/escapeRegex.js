const escapeRegex = text => {
  if (text !== "" || text !== null || text !== undefined) {
    // replace all white space with +, globally (replace all occurrences)
    return text.replace(/[\s]/g, "+");
  }
};

export default escapeRegex;
