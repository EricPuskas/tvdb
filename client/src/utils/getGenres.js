import React from "react";
import uuid from "uuid/v4";

const getGenres = genres => {
  if (!genres) return null;
  return genres.map(genre => (
    <li key={uuid()}>
      <span className="tag">{genre}</span>
    </li>
  ));
};

export default getGenres;
