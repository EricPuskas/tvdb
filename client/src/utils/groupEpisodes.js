const groupEpisodes = data => {
  let seasons = [];
  for (let i = 0; i < data.length; i++) {
    if (!seasons[data[i].airedSeason]) seasons[data[i].airedSeason] = [];
    seasons[data[i].airedSeason].push({
      id: data[i].id,
      season: data[i].airedSeason,
      episode: data[i].airedEpisodeNumber,
      title: data[i].episodeName
    });
  }
  return seasons;
};

export default groupEpisodes;
