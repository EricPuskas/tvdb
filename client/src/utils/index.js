import { applyMiddleware, createStore } from "redux";
import rootReducer from "../reducers";
import { middleware } from "../store";

export const findByAttribute = (component, attr) =>
  component.find(`[data-test="${attr}"]`);

export const testStore = initState => {
  const createStoreWithMiddleware = applyMiddleware(...middleware)(createStore);
  return createStoreWithMiddleware(rootReducer, initState);
};
