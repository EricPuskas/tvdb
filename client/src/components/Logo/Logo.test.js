import React from "react";
import { mount } from "enzyme";
import { findByAttribute } from "../../utils";
import { BrowserRouter } from "react-router-dom";
import Logo from "./Logo";

describe("SearchInput Component", () => {
  describe("Renders with props", () => {
    const props = {
      text: "test"
    };
    const component = mount(
      <BrowserRouter>
        <Logo {...props} />
      </BrowserRouter>
    );
    const wrapper = findByAttribute(component, "logo");
    const logoText = findByAttribute(component, "logo-text");

    it("Should pass snapshot test", () => {
      expect(component.html()).toMatchSnapshot();
    });

    it("Should render without errors", () => {
      expect(wrapper.length).toBe(1);
    });

    it("Should render text", () => {
      expect(logoText.length).toBe(1);
    });
  });
  describe("Renders with no props", () => {
    const component = mount(
      <BrowserRouter>
        <Logo />
      </BrowserRouter>
    );
    const wrapper = findByAttribute(component, "logo");
    const logoText = findByAttribute(component, "logo-text");

    it("Should pass snapshot test", () => {
      expect(component.html()).toMatchSnapshot();
    });

    it("Should render without errors", () => {
      expect(wrapper.length).toBe(1);
    });

    it("Should render text", () => {
      expect(logoText.length).toBe(0);
    });
  });
});
