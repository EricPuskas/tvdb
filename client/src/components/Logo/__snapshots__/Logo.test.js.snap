// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput Component Renders with no props Should pass snapshot test 1`] = `"<div class=\\"Logo\\" data-test=\\"logo\\"><a href=\\"/\\"><img src=\\"https://res.cloudinary.com/zenipsstudio/image/upload/v1572047365/logo.png\\" alt=\\"TVDB\\"></a></div>"`;

exports[`SearchInput Component Renders with props Should pass snapshot test 1`] = `"<div class=\\"Logo\\" data-test=\\"logo\\"><a href=\\"/\\"><img src=\\"https://res.cloudinary.com/zenipsstudio/image/upload/v1572047365/logo.png\\" alt=\\"TVDB\\"></a><h2 data-test=\\"logo-text\\"> test </h2></div>"`;
